pkgs/development/compilers/gcc/common/external-bootstrap.nix

```diff
@@ -0,0 +1,166 @@
+{ lib
+, stdenv
+, nukeReferences
+, version
+, langC
+, langCC
+, langJit
+, enablePlugin
+, profiledCompiler
+, enableExternalBootstrap
+, enableLibGccOutput
+}:
+
```

Comment:
This file needs a comment what it tries to achieve. Does it just build gcc?

Reply:
> This file needs a comment what it tries to achieve. Does it just build gcc?

It contains 99% of this PR's changes to `gcc`.

It exists in order to avoid copying and pasting the same block of code into multiple `gcc/${version}/default.nix` files.  Those are already a minefield every time we make a change to the duplicated code blocks.